### ISSUE TITLE
podman can not start after reboot

### ISSUE THREAD

**Issue by hqch0708** [containers/podman#19144]
### Issue Description

when Mac M1 was reboot, podman can not be used anymore. I think the root cause that is when reboot, podman did not update the status of machine.

The workaround is to update your **starting** to **false** in "/Users/{User}/.config/containers/podman/machine/qemu/podman-machine-default.json", and then podman can be worked as expection

### Steps to reproduce the issue

Steps to reproduce the issue
1. reboot 
2. podman machine start
cannot start VM podman-machine-default: VM already running or starting
3. podman machine stop
Machine "podman-machine-default" stopped successfully
4. podman machine list
podman-machine-default*  qemu        2 days ago  Currently starting  1           2.147GB     107.4GB

### Describe the results you received

cannot start VM podman-machine-default: VM already running or starting

### Describe the results you expected

podman can work

### podman info output

```yaml
Host:
  Arch: arm64
  CurrentMachine: podman-machine-default
  DefaultMachine: podman-machine-default
  EventsDir: /var/folders/_c/w5h26y1x6ksc3ndly5027d7h0000gn/T/podman-run--1/podman
  MachineConfigDir: /Users/{User}/.config/containers/podman/machine/qemu
  MachineImageDir: /Users/{User}/.local/share/containers/podman/machine/qemu
  MachineState: Starting
  NumberOfMachines: 1
  OS: darwin
  VMType: qemu
Version:
  APIVersion: 4.5.1
  Built: 1685113812
  BuiltTime: Fri May 26 23:10:12 2023
  GitCommit: 9eef30051c83f62816a1772a743e5f1271b196d7
  GoVersion: go1.20.4
  Os: darwin
  OsArch: darwin/arm64
  Version: 4.5.1
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by hqch0708** [comment]
BYW, I did not install podman-mac-helper at that moment

**Comment by Luap99** [comment]
If you remove the machine and then init a new one it should work again.

@vrothberg @baude @ashley-cui PTAL, seems like you can get the machine stucked in starting start even after a reboot.

**Comment by hqch0708** [comment]
@Luap99  yes, remove and init a new one is an easy way, but in this way, the existing containers and even images will be removed, so I think this way is not anyone wants to use.

**Comment by ashley-cui** [comment]
This should be fixed in Podman v4.6: https://github.com/containers/podman/pull/18328
